pipeline/trusted-resources/pkg/webhook/validation_admit_test.go

```diff
@@ -0,0 +1,364 @@
+package validation
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	imgname "github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	v1types "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/cosign/pkg/cosign"
+	sigs "github.com/sigstore/cosign/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	faketekton "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/fake"
+	remotetest "github.com/tektoncd/pipeline/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/webhook/json"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+var (
+	nameSpace      = "tekton-pipelines"
+	serviceAccount = "tekton-pipelines-webhook"
+)
+
+func Test_verifyResources(t *testing.T) {
```

Comment:
It's useful to have tests for both individual funcs (e.g. just the signature generation, etc.) as well as the overall verification. This way it's easier to figure out where exactly changes have been made if/when something breaks.

Reply:
sure!